### PR TITLE
Include task publishPluginZipPublicationToMavenLocal

### PR DESCRIPTION
**Pull request by prudhvigodithi**
### Description
Include task publishPluginZipPublicationToMavenLocal
 
### Issues Resolved
https://github.com/opensearch-project/job-scheduler/issues/583
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by prudhvigodithi** [comment]
Adding @owaiskazi19

File: settings.gradle [hunk 1]
```diff
@@ -10,4 +10,4 @@ project(":spi").name = rootProject.name + "-spi"
 
 include "sample-extension-plugin"
 project(":sample-extension-plugin").name = rootProject.name + "-sample-extension"
-startParameter.excludedTaskNames=["publishPluginZipPublicationToMavenLocal", "publishPluginZipPublicationToStagingRepository"]
+startParameter.excludedTaskNames=["publishPluginZipPublicationToStagingRepository", "bwcTestSuite"]
```
Comment (owaiskazi19) [inline settings.gradle:13]: Why are we adding `bwcTestSute' here?

**Comment by owaiskazi19** [comment]
@prudhvigodithi are you able to see jars and zips now for job-scheduler and spi now?

**Comment by prudhvigodithi** [comment]
I see the artifacts in right locations. 
#### Regular Zip plugin:
`tar/builds/opensearch/plugins/opensearch-job-scheduler-2.12.0.0.zip`

#### Staging Maven repo:
`tar/builds/opensearch/maven/org/opensearch/opensearch-job-scheduler/2.12.0.0`
`tar/builds/opensearch/maven/org/opensearch/opensearch-job-scheduler-spi/2.12.0.0`
`tar/builds/opensearch/maven/org/opensearch/plugin/opensearch-job-scheduler/2.12.0.0`

#### Local Maven repo:
`.m2/repository/org/opensearch/opensearch-job-scheduler/2.12.0.0`
`.m2/repository/org/opensearch/opensearch-job-scheduler-spi/2.12.0.0`
`.m2/repository/org/opensearch/opensearch-job-scheduler-sample-extension/2.12.0.0`
`.m2/repository/org/opensearch/plugin/opensearch-job-scheduler/2.12.0.0`


Adding @bbarani @vibrantvarun  @peterzhuamazon @cwperks @dbwiddis.

**Review by peterzhuamazon — APPROVED** [review]

**Review by joshpalis — APPROVED** [review]

**Comment by prudhvigodithi** [comment]
Raised a manual backport PR https://github.com/opensearch-project/job-scheduler/pull/585, please check.

**Pull request merged**